### PR TITLE
admin/user route fixes

### PR DESCRIPTION
**Pull request by zulqarnainhanif**
## Summary

- fix user fetching query
- use common table and confirmation component for deletion
- update UI for role updating popup and view user UI
- fix crash on delete user and and user name in delete popup
- allow search with name of user
- fix crash on create user
- fix user runtime role creation flow

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [x] If this PR is still a WIP, convert to a draft 
- [x] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

https://github.com/XRFoundation/XREngine/issues/4607
https://github.com/XRFoundation/XREngine/issues/4805
https://github.com/XRFoundation/XREngine/issues/4932
https://github.com/XRFoundation/XREngine/issues/4942
https://github.com/XRFoundation/XREngine/issues/4943

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

@HexaField @speigg @NateTheGreatt

### CODE REVIEW

File: packages/client-core/src/admin/components/Users/ViewUser.tsx [hunk 1]
```diff
@@ -346,33 +358,37 @@ const ViewUser = (props: Props) => {
...
-                  disableCloseOnSelect
-                  filterOptions={(options) =>
-                    options.filter(
-                      (option) => state.scopeTypes.find((scopeType) => scopeType.type === option.type) == null
-                    )
-                  }
-                  getOptionLabel={(option) => option.type}
-                  renderInput={(params) => <TextField {...params} placeholder="Select scope" />}
-                />
+                <FormControl fullWidth>
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Users/ViewUser.tsx:363]: @speigg this select component

**Comment by zulqarnainhanif** [comment]
@kimenyikevin just mentioning the tasks here for you:

https://github.com/XRFoundation/XREngine/issues/4942
https://github.com/XRFoundation/XREngine/issues/4943

**Comment by zulqarnainhanif** [comment]
@kimenyikevin use the package `@speigg/hookstate` instead of `@hookstate/core`

**Review by kimenyikevin — APPROVED** [review]

**Comment by zulqarnainhanif** [comment]
@kimenyikevin please fix the type error in CreateUser and ViewUser indicated by the pipeline.

**Comment by zulqarnainhanif** [comment]
@kimenyikevin alert goes behind the top header

Screenshot 2022-01-30 at 10 26 28 PM

**Comment by zulqarnainhanif** [comment]
after updating user role the value doesn't update in the sidepanel drawer and the table behind in the view (manual refresh is required)
Screenshot 2022-01-30 at 10 27 04 PM

**Comment by zulqarnainhanif** [comment]
@kimenyikevin user shouldn't see delete button in front of his/her own account, could accidentally lose access.

**Comment by kimenyikevin** [comment]
> @zulqarnainhanif Should I remove access to edit user role and change user scope for user who logged in

**Comment by zulqarnainhanif** [comment]
> 

edit user access is fine.

**Comment by zulqarnainhanif** [comment]
> > @zulqarnainhanif Should I remove access to edit user role and change user scope for user who logged in

Edit user access is fine

**Pull request merged**